### PR TITLE
Add a couple of tests for SemanticModel.GetTypeInfo(PatternSyntax)

### LINKED ISSUE #26494: GetTypeInfo for a PatternSyntax
GetTypeInfo for a PatternSyntax should give the incoming type, and possibly also the type required for matching (ConvertedType). For example in the code
``` c#
bool IsThree(object o) => o is 3;
```
The result of `GetTypeInfo` for pattern `3` should have `Type` of `object`, and a `ConvertedType` of `int` (the incoming object is converted to int for the match).

### PR DESCRIPTION
**Pull request by gafter**
Fixes #26494

@cston @agocke Please review this modest mostly-test addition for pattern-matching
/cc @dotnet/roslyn-compiler

### CODE REVIEW

**Comment by gafter** [comment]
@cston Adjusted as discussed.

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs [hunk 1]
```diff
@@ -2151,5 +2151,165 @@ public class C {
...
+
+        [Fact]
+        public void PatternTypeInfo_03()
+        {
+            var source = @"
+public class C
+{
+    void M(object o)
+    {
+        if (o is Point(3, 4.0) { Missing: Xyzzy }) {}
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs]: >Point( [](start = 17, length = 6)

Unrelated to this PR, the current formatting rule is to put a space between type and positional sub-pattern. Please file an issue if you think it should be formatted like an invocation instead (ie. no space).

File: src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs [hunk 2]
```diff
@@ -1872,12 +1872,14 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
             BoundNode highestBoundNode,
             BoundNode boundNodeForSyntacticParent)
         {
-            if (lowestBoundNode is BoundPattern pattern)
+            BoundPattern pattern = lowestBoundNode as BoundPattern ?? highestBoundNode as BoundPattern ?? (highestBoundNode is BoundSubpattern sp ? sp.Pattern : null);
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs]: Is there a case where `lowestBoundNode` is a BoundSubpattern, or is that case never useful in practice (already handled by other cases)?

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs (hunk 1 above)
Comment (gafter) [reply src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs]: https://github.com/dotnet/roslyn/issues/27268

File: src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs (hunk 2 above)
Comment (gafter) [reply src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs]: I think the bound pattern would always be lower than a bound subpattern, as the API rejects a subpattern syntax as input.

**Pull request merged**